[docs] Document the `PickerValidDate` type override (#13476)
mui/mui-x: docs/data/migration/migration-pickers-v6/migration-pickers-v6.md

 />
```

### Strict typing of the date-related props

All the date-related props are now strictly typed to only accept the date format supported by your adapter
(`Date` object for `date-fns`, `daysjs.Dayjs` object for `days-js`, etc.).

> You can learn more about this change in the [dedicated doc section](/x/react-date-pickers/base-concepts/#typing-of-the-date).

Reviewer: You can also consider using the format suggested by Sam on how to name the links to [other doc sections](https://mui.com/x/react-date-pickers/date-field/#customize-the-date-format).
```suggestion
:::info
See [Base concepts—Typing of the date](/x/react-date-pickers/base-concepts/#typing-of-the-date) for more details.
:::
```

Author: Would you apply the same rational to the other "You can learn more" of the page?

Reviewer: It's just a suggestion, but especially in this section, I think that it would improve the visibility and importance of the link.
As for the other sections—it's up to you. Technically it's out of the scope of this issue, but I don't think it would hurt. 👌

Author: It's out of the scope but it's a super simple change that is not worth a dedicated review IMHO, I'll do the switch :ok_hand:

Author: Done :+1:

Author: I will do a follow up for the rest of the doc if I find some time

 />
```

### Strict typing of the date-related props

All the date-related props are now strictly typed to only accept the date format supported by your adapter

> (`Date` object for `date-fns`, `daysjs.Dayjs` object for `days-js` etc...).

Reviewer: ```suggestion
(`Date` object for `date-fns`, `daysjs.Dayjs` object for `days-js`, etc.).
```
https://www.grammarly.com/blog/et-cetera-etc/

> If you run into TypeScript errors such as `DesktopDatePickerProps<Date> error Type 'Date' does not satisfy the constraint 'never'`,

Author: I made sure we have the error here so that people using `Ctrl + F` can easily find the section in the migration guide.

Reviewer: The page has a search function as well.
WDYT about making the section in this page a bit concise? 🤔 
As it currently stands, it seems that there is little point in linking to the `base-concepts` page, because this section contains almost all the same information. 🙈 🤷

Author: Whay would you remove from this section?

Author: Done :+1:

The `dateLibInstance` prop of `LocalizationProvider` does not work with `AdapterDayjs` anymore.
This prop was used to set the pickers in UTC mode before the implementation of a proper timezone support in the components.

> Learn more about the new approach on the [dedicated doc page](https://mui.com/x/react-date-pickers/timezone/).

Reviewer: ```suggestion
Learn more about the new approach on the [dedicated doc page](/x/react-date-pickers/timezone/).
```
Here Grammarly suggests the previous usage (You can learn) though. 🤔

Author: We had 2 "Look" and 1 "You can look" so I took the most common
But if Grammarly prefers "You can look" I will use that one :ok_hand:
